### PR TITLE
added dynamoDB functionality

### LINKED ISSUE #156: Integrate DynamoDB into PC MVP

### LINKED ISSUE #173: Integrate DynamoDB to AWS cloud

### PR DESCRIPTION
**Pull request by andrewnguyen22**
Note: The DB tests won't run correctly without the database being built. I will look into the best CI configuration to make this work well. I predict that the CI will need aws credentials and an installed version of dynamodb

closes #156 
EDIT:

Now closes #156 and #155 

EDIT2:

Now closes #173

### CODE REVIEW

File: cmd/db/dynamodb.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	_, err := fmt.Scanf("%d", &i)
+	if err != nil {
+		fmt.Println(err)
+	}
+	switch i {
+	case 1:
+		CreateTable()
+	case 2:
+		DeleteTable()
+	}
```
Comment (iajrz) [inline cmd/db/dynamodb.go:57]: This behaves better as part of a while loop, starting at about line 42, and with a default case that would advise the user to choose either 1 or 2 depending on the situation and restart.

File: rpc/client/live.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ import (
 func Register(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	n := &node.Node{}
 	if err := shared.PopModel(w, r, ps, n); err != nil {
-		shared.WriteJSONResponse(w, "500 error: "+err.Error())
+		shared.WriteErrorResponse(w, 500, err.Error())
```
Comment (iajrz) [inline rpc/client/live.go]: I think this is more likely a 400 Bad Request

File: rpc/client/live.go [hunk 3]
```diff
@@ -25,14 +25,14 @@ func Register(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
...
-	shared.WriteJSONResponse(w, "Invalid credentials")
+	shared.WriteErrorResponse(w, 401, "Invalid credentials")
 }
 
 // "Register" handles the localhost:<client-port>/v1/register call.
 func UnRegister(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	n := &node.Node{}
 	if err := shared.PopModel(w, r, ps, n); err != nil {
-		shared.WriteJSONResponse(w, "500 error: "+err.Error())
+		shared.WriteErrorResponse(w, 500, err.Error())
```
Comment (iajrz) [inline rpc/client/live.go]: I think this is more likely a 400 Bad Request

File: util/ip.go [hunk 4]
```diff
@@ -1,14 +1,12 @@
...
 
 import (
-	"fmt"
 	"io/ioutil"
 	"net/http"
 )
 
 func IP() (string, error) {
 	url := "https://api.ipify.org?format=text"
-	fmt.Printf("Getting IP address from  ipify ...\n")
```
Comment (iajrz) [inline util/ip.go:11]: Maybe this could be an Info level log instead of not existing anymore

File: cmd/db/dynamodb.go (hunk 1 above)
Comment (andrewnguyen22) [reply cmd/db/dynamodb.go:57]: Agreed ^ will fix this

File: rpc/client/live.go (hunk 2 above)
Comment (andrewnguyen22) [reply rpc/client/live.go]: Yes agreed, a badly formatted request should be the only reason for error. Will fix this

File: util/ip.go (hunk 4 above)
Comment (andrewnguyen22) [reply util/ip.go:11]: Yes good thought!

**Comment by andrewnguyen22** [comment]
Requesting a new review! @iajrz @luyzdeleon

File: cmd/db/dynamodb.go [hunk 5]
```diff
@@ -39,16 +40,20 @@ func DB() *dynamodb.DynamoDB {
...
+		_, err := fmt.Scanf("%d", &i)
+		if err != nil {
+			fmt.Println(err)
+		}
+		switch i {
+		case 1:
+			CreateTable()
+		case 2:
+			DeleteTable()
+		default:
```
Comment (iajrz) [inline cmd/db/dynamodb.go]: this does essentially the same as not having the loop; I meant to have a loop such that allowed you to "try again" if you made a mistake in the input (say, pressed '12' or '`' by mistake.)

Comment (andrewnguyen22) [reply cmd/db/dynamodb.go]: Oh is that what you meant? I don't really see the need for that kind of error checking?

Comment (andrewnguyen22) [reply cmd/db/dynamodb.go]: This loop allows you to create and delete the table in one running session.

Comment (andrewnguyen22) [reply cmd/db/dynamodb.go]: In any case ^ I have updated the code to reflect the result you desired

**Review by iajrz — DISMISSED** [review]

**Pull request merged**